### PR TITLE
👽️ Connect Dashboard Dropdown Filters to New Aggregations (#2046)

### PR DESCRIPTION
**Pull request by mistryrn**
**Description of changes**

* Update Program Donor Summary GQL query to fetch new filter count aggregations
* Fetch filter counts and display them beside each option on dashboard table dropdown filters
* Fix a bug in DropdownPanel component preventing the number `0` from appearing as a stylized count beside an option

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx [hunk 1]
```diff
@@ -378,6 +381,46 @@ export default ({
     );
   };
 
+  const [pagingState, setPagingState] = React.useState({
```
Comment (mistryrn) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx:384]: Switcheroo pt 1: line 324-423 here are not new code, I just had to move this block around to make the linter happy

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx [hunk 2]
```diff
@@ -581,46 +672,6 @@ export default ({
     },
   ];
 
-  const [pagingState, setPagingState] = React.useState({
```
Comment (mistryrn) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx:584]: Switcheroo pt 2: lines 584-623 is where the block of code from pt 1 used to be

File: uikit/DropdownPanel/index.tsx [hunk 3]
```diff
@@ -440,7 +440,7 @@ export const ListFilter = ({
                       `}
                     />
                     {option.value}
-                    {option.doc_count && (
+                    {typeof option?.doc_count === 'number' && (
```
Comment (mistryrn) [inline uikit/DropdownPanel/index.tsx:443]: When `doc_count` was `0` it was short-circuiting here and you'd just get an un-styled `0` beside the option value. Checking for type number now to make sure even `0` gets styled

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx (hunk 2 above)
Comment (caravinci) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx:584]: I see what you did there, and I like it!

**Review by caravinci — APPROVED** [review]
:shipit: 

---
Side note: I love how some of us are starting to consistently comment through our PRs. 💪

**Pull request merged**